src/AnalysisPrograms/Recognizers/GenericRecognizer.cs

```diff
@@ -135,12 +135,55 @@ public static void ValidateProfileTagsMatchAlgorithms(Dictionary<string, object>
             var results = RunProfiles(audioRecording, configuration, segmentStartOffset);
 
             // ############################### POST-PROCESSING OF GENERIC EVENTS ###############################
+
             var postprocessingConfig = configuration.PostProcessing;
-            results.NewEvents = EventPostProcessing.PostProcessingOfSpectralEvents(
-                results.NewEvents,
-                postprocessingConfig,
-                results.Sonogram,
-                segmentStartOffset);
+            var postEvents = new List<EventCommon>();
+
+            // count number of events detected at each decibel threshold.
+            for (int i = 1; i <= 39; i++)
+            {
+                var dbEvents = EventFilters.FilterOnDecibelDetectionThreshold(results.NewEvents, (double)i);
+
+                if (dbEvents.Count > 0)
+                {
+                    //Log.Debug($"Profiles detected {dbEvents.Count} events at threshold {i} dB.");
+                    var ppEvents = EventPostProcessing.PostProcessingOfSpectralEvents(
+                        dbEvents,
+                        postprocessingConfig,
+                        (double)i,
+                        results.Sonogram,
+                        segmentStartOffset);
+
+                    postEvents.AddRange(ppEvents);
+                }
+            }
+
+            // Running profiles with multiple dB thresholds produces nested (Russian doll) events.
+            // Remove all but the outermost event.
+            // Add a spacer for easier reading of the debug output.
+            Log.Debug($" ");
+            Log.Debug($"Event count BEFORE removing enclosed events = {postEvents.Count}.");
+            results.NewEvents = CompositeEvent.RemoveEnclosedEvents(postEvents);
```

Comment:
This comment remains unanswered?

Reply:
I think I understand, this is not flattening nested events, or even flattening Russian dolls, rather this is removing any event that is completely overlapped in time/frequency bounds by another event covering the same area.

If my new understanding is correct, then no change is required, other than updating the log message. 

The terminology nested is wrong here since there is no parent-child relationship. Hence the confusion.